### PR TITLE
Add wording about keeping PRs up-to-date in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by yardenshoham**
Following a discord discussion from today I've written the policy explained to me about keeping PRs up-to-date.

### CODE REVIEW

**Comment by yardenshoham** [comment]
Please make sure what I've written is true

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -146,6 +146,14 @@ If your PR could cause a breaking change you must add a BREAKING section to this
 
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase your branch as it makes it
```
Comment (wxiaoguang) [inline CONTRIBUTING.md]:
```suggestion
Once code review starts on your PR, do not rebase or squash your branch as it makes it
```

**Review by wxiaoguang — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -146,6 +146,14 @@ If your PR could cause a breaking change you must add a BREAKING section to this
 
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase or squash your branch as it makes it
+difficult to review your code. Only if there is a need, sync your branch by merging
```
Comment (wxiaoguang) [inline CONTRIBUTING.md]:
```suggestion
difficult to review the new changes. Only if there is a need, sync your branch by merging
```

**Review by 6543 — APPROVED** [review]
ah thanks!!! 

we should long ago have put that in

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -146,6 +146,14 @@ If your PR could cause a breaking change you must add a BREAKING section to this
 
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase nor squash your branch as it makes it
+difficult to review the new changes. Only if there is a need, sync your branch by merging
+the base branch into yours. Don't worry about merge commits messing up your tree as
+the final merge process squashes all commits into one, using the description of the PR's first comment.
```
Comment (6543) [inline CONTRIBUTING.md]:
```suggestion
the final merge process squashes all commits into one, using the description of the Pull.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -146,6 +146,14 @@ If your PR could cause a breaking change you must add a BREAKING section to this
 
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase nor squash your branch as it makes it
+difficult to review the new changes. Only if there is a need, sync your branch by merging
+the base branch into yours. Don't worry about merge commits messing up your tree as
+the final merge process squashes all commits into one, using the description of the Pull.
```
Comment (6543) [inline CONTRIBUTING.md]:
```suggestion
the final merge process squashes all commits into one, using the description of the PR.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -146,6 +146,14 @@ If your PR could cause a breaking change you must add a BREAKING section to this
 
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase nor squash your branch as it makes it
+difficult to review the new changes. Only if there is a need, sync your branch by merging
+the base branch into yours. Don't worry about merge commits messing up your tree as
+the final merge process squashes all commits into one, using the description of the PR.
```
Comment (6543) [inline CONTRIBUTING.md]: if you want you we could make that even more clear ...

the visible commit message (first line) is the pull title + index and  the pull description gets appended

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -146,6 +146,15 @@ If your PR could cause a breaking change you must add a BREAKING section to this
...
 To explain how this could affect users and how to mitigate these changes.
 
+Once code review starts on your PR, do not rebase nor squash your branch as it makes it
+difficult to review the new changes. Only if there is a need, sync your branch by merging
+the base branch into yours. Don't worry about merge commits messing up your tree as
+the final merge process squashes all commits into one, with the visible commit message (first
+line) being the PR title + PR index and description being the PR's first comment.
+
+Once your PR gets the `lgtm/done` label, don't worry about keeping it up-to-date or breaking builds.
+It is the maintainer team's responsibility from this point to get it merged.
```
Comment (techknowlogick) [inline CONTRIBUTING.md]: A tiny nitpick, sometimes if it is conflicting a maintainer will ask submitter to resolve the conflicts prior to merging

**Pull request merged**